src/libraries/Microsoft.Extensions.Caching.Memory/tests/CacheEntryScopeExpirationTests.cs

```diff
@@ -62,9 +62,9 @@ public void SetPopulates_AbsoluteExpiration_IntoScopeLink()
                 cache.Set(key, obj, new MemoryCacheEntryOptions().SetAbsoluteExpiration(time));
             }
 
-            Assert.Null(((CacheEntry)entry)._expirationTokens);
-            Assert.NotNull(((CacheEntry)entry)._absoluteExpiration);
-            Assert.Equal(time, ((CacheEntry)entry)._absoluteExpiration);
+            Assert.Empty(((CacheEntry)entry).ExpirationTokens);
```

Comment:
Would it be better to ensure this list is not created at all above? We could do this either by:

* leaving the field `internal` and keeping the same code here
* Using Reflection to get the value of the field

Reply:
I am not sure what would be the best and what was the intention here.

The test creates an expiration token and does not use it at all (this is surprising to me).
It asserts that the token was not added to cache entry expiration tokens (in some implicit way, from a scope??).

I prefer to test behavior, not implementation and I would keep it as it is in this PR (asserting that public property returns an empty collection).

@eerhardt what do you think?

Reply:
> The test creates an expiration token and does not use it at all (this is surprising to me).

I assume that is just a copy-paste left over from the test above it.

> I prefer to test behavior, not implementation and I would keep it as it is in this PR (asserting that public property returns an empty collection).

My thinking was that before we ensured a new List object wasn't needlessly created (for example by some code accessing the property to initialize it). But now by going through the property in the test, we lost this assurance.

I'm fine with leaving it the way you have it. Just wanted to be sure this loss in testing was acceptable.